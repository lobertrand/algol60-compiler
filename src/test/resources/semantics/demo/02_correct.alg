begin
    comment Array as parameter of procedure;
    integer procedure max(arr, a, b);
    value arr, a, b; integer array arr; integer a, b;
    begin
        integer i, result;
        result := arr[a];
        comment For-statement;
        for i := a+1 step 1 until b do
        begin
            comment If-statement;
            if arr[i] > result then
                result := arr[i]
        end;
        max := result
    end;

    comment Array declaration and assignment;
    integer array a[1:3];
    a[1] := 6;
    a[2] := 8;
    a[3] := 3;

    comment Procedure call;
    outinteger(1, max(a, 1, 3))
end
