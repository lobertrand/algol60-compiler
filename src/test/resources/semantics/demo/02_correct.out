Algol60> Lexical analysis successful
Algol60> Syntactic analysis successful
┌───────────────────────────────────────────────┐
│ level: 0, tableNumber: 0                      │
├───────────────────────────────────────────────┤
│ Procedure: void outstring(integer, string)    │
│ Procedure: void outboolean(integer, boolean)  │
│ Procedure: void outinteger(integer, integer)  │
│ Procedure: void outreal(integer, real)        │
│ Procedure: void inreal(integer, real)         │
│ Procedure: void ininteger(integer, integer)   │
│ Procedure: integer entier(real)               │
└───────────────────────────────────────────────┘
    ┌──────────────────────────────────────────────────────────┐
    │ level: 1, tableNumber: 1                                 │
    ├──────────────────────────────────────────────────────────┤
    │ Procedure: integer max(integer array, integer, integer)  │
    │ Array: integer array a[1:3]                              │
    └──────────────────────────────────────────────────────────┘
        ┌──────────────────────────────────────────┐
        │ level: 2, tableNumber: 2                 │
        ├──────────────────────────────────────────┤
        │ Array: integer array arr[] (value param) │
        │ Variable: integer a (value param)        │
        │ Variable: integer b (value param)        │
        │ Variable: integer max (proc. result)     │
        │ Variable: integer i                      │
        │ Variable: integer result                 │
        └──────────────────────────────────────────┘
            ┌──────────────────────────┐
            │ level: 3, tableNumber: 3 │
            ├──────────────────────────┤
            └──────────────────────────┘

Algol60> Semantic analysis successful
